core/src/io/anuke/mindustry/entities/type/TileEntity.java

```diff
@@ -307,14 +307,17 @@ public void update(){
         }
 
         Block previous = block;
-        block.update(tile);
-        if(block == previous && cons != null){
+        if(cons != null){
             cons.update();
         }
 
-        if(block == previous && power != null){
-            power.graph.update();
+        float amountOfPowerUsed = 0.0f;
+        
+        if(power != null){
+            amountOfPowerUsed = power.graph.update();
```

Comment:
This shouldn't be needed if power is queried in the block - don't change `TileEntity`.